zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -1,34 +1,59 @@
 package edu.uci.ics.cloudberry.zion.model.impl
 
-import edu.uci.ics.cloudberry.zion.model.datastore.{IJSONParser, JsonRequestException}
+import edu.uci.ics.cloudberry.zion.model.datastore.{FieldNotFound, IJSONParser, JsonRequestException, QueryParsingException}
 import edu.uci.ics.cloudberry.zion.model.schema.Relation.Relation
 import edu.uci.ics.cloudberry.zion.model.schema._
 import play.api.libs.functional.syntax._
 import play.api.libs.json.{JsObject, _}
 
+
+/**
+  * Parse a [[JsValue]] with a schema map, and returns a [[Query]]
+  * First parse the [[JsValue]] into a [[UnresolvedQuery]] using json parsing APIs provided by Play framework.
+  * Then, calls [[QueryResolver]] to resolve it into a [[Query]], which has all fields resolved typed.
+  * Finally, calls [[QueryValidator]] to validate the correctness of this query.
+  */
 class JSONParser extends IJSONParser {
 
   import JSONParser._
 
-  override def parse(json: JsValue): (Seq[Query], QueryExeOption) = {
+  override def getDatasets(json: JsValue): Set[String] = {
+    val datasets = (json \\ "dataset").filter(_.isInstanceOf[JsString]).map(_.asInstanceOf[JsString].value)
+    if (datasets.isEmpty) {
+      datasets.toSet
+    } else {
+      //TODO currently do not handle lookup queries
+      Set(datasets.head)
+    }
+  }
+
+  override def parse(json: JsValue, schemaMap: Map[String, Schema]): (Seq[Query], QueryExeOption) = {
     val option = (json \ "option").toOption.map(_.as[QueryExeOption]).getOrElse(QueryExeOption.NoSliceNoContinue)
     val query = (json \ "batch").toOption match {
-      case Some(groupRequest) => groupRequest.validate[Seq[Query]] match {
-        case js: JsSuccess[Seq[Query]] => js.get
+      case Some(groupRequest) => groupRequest.validate[Seq[UnresolvedQuery]] match {
+        case js: JsSuccess[Seq[UnresolvedQuery]] => js.get
         case e: JsError => throw JsonRequestException(JsError.toJson(e).toString())
       }
-      case None => json.validate[Query] match {
-        case js: JsSuccess[Query] => Seq(js.get)
+      case None => json.validate[UnresolvedQuery] match {
+        case js: JsSuccess[UnresolvedQuery] => Seq(js.get)
         case e: JsError => throw JsonRequestException(JsError.toJson(e).toString())
       }
     }
-    (query, option)
+    val resolvedQuery = query.map(resolve(_, schemaMap))
+    (resolvedQuery, option)
   }
 }
 
 object JSONParser {
   //Warn: the order of implicit values matters. The dependence should be initialized earlier
 
+  def resolve(query: UnresolvedQuery, schemaMap: Map[String, Schema]): Query = {
```

Comment:
For scala doc for `acceptType` and `resultType`, I added them in `IFunction` trait.